### PR TITLE
[CARBONDATA-3663] Support loading stage files in batches

### PR DESCRIPTION
**Pull request by niuge01**
### Why is this PR needed?
 When there are a lots of stage files in the stage directory,  if load all of them in once time, the loading time will can not be control.
There need a way for users to specify the number of stage files per processing, to control the execution time of commands.
 
 ### What changes were proposed in this PR?
Add a load option batch_file_count for users to specify the number of stage files per processing.
    
 ### Does this PR introduce any user interface change?
 - Yes

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1628/

File: integration/flink/src/test/scala/org/apache/carbon/flink/TestCarbonWriter.scala [hunk 1]
```diff
@@ -103,14 +103,91 @@ class TestCarbonWriter extends QueryTest {
...
+      try environment.execute
+      catch {
+        case exception: Exception =>
+          // TODO
+          throw new UnsupportedOperationException(exception)
+      }
+
+      sql(s"INSERT INTO $tableName STAGE OPTIONS ('batch_file_count' = '5')")
+
+      checkAnswer(sql(s"SELECT count(1) FROM $tableName"), Seq(Row(500)))
```
Comment (Zhangshunyu) [inline integration/flink/src/test/scala/org/apache/carbon/flink/TestCarbonWriter.scala]: why each stage file have 100 lines?

**Comment by Zhangshunyu** [comment]
LGTM

File: docs/dml-of-carbondata.md [hunk 2]
```diff
@@ -316,12 +316,14 @@ CarbonData DML statements are documented here,which includes:
   You can use this command to insert them into the table, so that making them visible for query.
   
   ```
-  INSERT INTO <CARBONDATA TABLE> STAGE
+  INSERT INTO <CARBONDATA TABLE> STAGE OPTIONS(property_name=property_value, ...)
```
Comment (jackylk) [inline docs/dml-of-carbondata.md]: please describe all supported options in this document

**Comment by niuge01** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1657/

File: integration/flink/src/test/scala/org/apache/carbon/flink/TestCarbonWriter.scala (hunk 1 above)
Comment (niuge01) [reply integration/flink/src/test/scala/org/apache/carbon/flink/TestCarbonWriter.scala]: Specified by the option CarbonLocalProperty.COMMIT_THRESHOLD.

writerProperties.put(CarbonLocalProperty.COMMIT_THRESHOLD, "100")

File: docs/dml-of-carbondata.md (hunk 2 above)
Comment (niuge01) [reply docs/dml-of-carbondata.md]: done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1671/

**Comment by niuge01** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1680/

**Pull request closed without merging**